src/binary/merkle_tree.rs

```diff
@@ -107,11 +106,21 @@ where
         side_positions.reverse(); // Reorder side positions from leaf to root.
         side_positions.pop(); // The last side position is the root; remove it.
 
+        // Allocate scratch storage to store temporary nodes when building the
+        // root.
+        let mut scratch_storage = StorageMap::<NodesTable>::new();
```

Comment:
Scratch storage is a bit of a confusing name here, since when reading I assumed it was a temporary buffer where values could be shuffled to and from within a method, not a storage place where a result value would be. 

A doc comment on `root_node` to indicate that it may write a desired value to the `scratch_space` would be nice since from a top of level it's not clear that `scratch_space` may contain result values